scripts/validate.sh

```diff
@@ -24,11 +25,21 @@ function validate_release_fields() {
     _validate 'release-notes'
     _validate 'components'
     for project in ${PROJECTS[*]}; do
-        _validate "components.${project}"
+        if _validate "components.${project}"; then
+            local commit_hash="${release["components.${project}"]}"
+            if [[ ! "$commit_hash" =~ ^[0-9a-f]+$ ]]; then
+                printerr "Commit hash of ${project} should only be hexadecimal: ${commit_hash}"
+                errors=$((errors+1))
+            fi
+            if [[ ${#commit_hash} != 40 ]]; then
+                printerr "Commit hash of ${project} should be 40 characters long: ${commit_hash}"
+                errors=$((errors+1))
+            fi
```

Comment:
This can be checked in a single test:

```suggestion
            if [[ ! "$commit_hash" =~ ^[0-9a-f]{40}$ ]]; then
                printerr "Commit hash of ${project} must be a 40-digit hexadecimal value: ${commit_hash}"
                errors=$((errors+1))
            fi
```